### PR TITLE
Add dispensersPlaceEyesOfEnder and dispensersRemoveEyesOfEnder carpet rules

### PR DESCRIPTION
**Pull request by Gilly7CE**
New rules:
- `dispensersPlaceEyesOfEnder`
- `dispensersRemoveEyesOfEnder`

Some refactoring done to reuse code from the `EndFrameBlock_EyeToggleMixin` class.

### CODE REVIEW

**Comment by Gilly7CE** [comment]
Will bump the mod version to 0.4.0 as part of this PR

File: src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     * @author serz
+     */
+    public static void setEmptyEndPortalFrameState(World world, BlockState blockState, BlockPos pos) {
+        BlockState newState = blockState.with(EndPortalFrameBlock.EYE, false);
+        world.setBlockState(pos, newState);
+        //remove portal this frame used to create
+        removePortal(world, pos.north());
+        removePortal(world, pos.west());
+        removePortal(world, pos.south());
+        removePortal(world, pos.east());
```
Comment (Serzbiedry) [inline src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java:47]: You can make it faster and more accurate by replacing quoted code with:
removePortal(world, pos.offset(state.get(FACING),1));

**Comment by Gilly7CE** [comment]
Adding rule has a bug where it removes 2 items from the stack not 1.

File: src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java (hunk 1 above)
Comment (Gilly7CE) [reply src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java]: Good suggestion! Refactored in latest changes.

File: src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public final class EndPortalFrameHelper {
+    /**
+     * Checks if the block state has an eye of ender
+     * @param blockState the block state to check
+     * @author Gilly7CE
+     */
+    public static boolean HasEyeOfEnder(BlockState blockState) {
```
Comment (Gilly7CE) [inline src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java:17]: Rename `hasEyeOfEnder`

Comment (Gilly7CE) [reply src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java]: Done.

**Review by Serzbiedry — APPROVED** [review]

**Pull request merged**